tests/e2e/instrumentation-service-name-annotation/00-install-instrumentation.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: java
+spec:
+  env:
```

Comment:
somebody can define OTEL_SERVICE_NAME here. What is the precedence order of setting the service name? We should document this in the readme and have a test.

Reply:
I honestly am not crazy sure on these answers. If you don't mind helping me fill the gaps.

Reply:
@blakeromano, I'm going to check what is missing to help you out.

Reply:
@yuriolisa I apologize been busy I was wondering if you had time to help me out 😸